src/collective/elasticindex/changes.py

```diff
@@ -71,7 +72,15 @@ def get_data(content, security=False, domain=None):
         return None, None
     title = content.Title()
     try:
-        text = content.SearchableText()
+        # wrap content, so the SearchableText will be 
+        # delegated to IIndexer adapters as appropriate.
+        # This will e.g. take care of PDF file indexing.
+        cat = getToolByName(content, 'portal_catalog')
+        wrapper = IndexableObjectWrapper(content, cat)
```

Comment:
Instead of hard coding the wrapper, retrieve it with queryMultiAdapter like it is done in Products/CMFPlone/CatalogTool.py:catalog_object
code not tested:

    from plone.indexer.interfaces import IIndexableObject
    cat = getToolByName(content, 'portal_catalog')
    wrapper = content                                                              
    if not IIndexableObject.providedBy(content):                             
        # This is the CMF 2.2 compatible approach, which should be used        
        # going forward                                                     
        w = queryMultiAdapter((content, cat), IIndexableObject)          
        if wrapper is not None:                                             
            wrapper = w
        text = getattr(wrapper, 'SearchableText', None)
        if safe_callable(text):
            text = text()